src/supported-operations/thin-edge-supported-operations.md

```diff
@@ -0,0 +1,109 @@
+# Supported operations
+
+## Support for operations
+
+Supported operations would be a part of tedge config.
+
+Supported operations are listed per supported cloud.
+
+Can be modified by hand or using tedge cli tool:
+
+* modify fields in `tedge.toml` file:
+  * proposed to use `operations` key in a cloud specific toml table
+  * the key is an `array` of `strings` with *literal values* for supported operations (no further modifications required by users, like prepending cumulocity operations with `c8y_`)
+  * that means that the key operations would exist in the cloud table like so (example for c8y):
+
+```toml
+[...]
+
+[c8y]
+operations = ["c8y_SoftwareUpdate", "...", ...]
+```
+
+Q: How to support parametrized operations / features for operations?
+A1: Maybe use operations name as allowed key in config and allow list of parameters
+A2: Maybe add prefix to the key from above.
+
+```toml
+[...]
+
+[c8y]
+operations = ["c8y_SoftwareUpdate", "...", ...]
+c8y_SoftwareUpdate = [""]
+```
+
+* use `tedge cli` command to add or remove operations one by one, list all operations, list all operations per cloud
+  * use new tedge subcommand `tedge operations`
+  * `tedge operations` supports following operations:
+    * `add` cloud_name operation_name - adds single operation to the list if doesn't exist
+    * `remove` cloud_name operation_name - removes single operation from the list if exists
+    * `list [cloud_name]` - lists all operations, unless specific cloud table name provided, then lists only operations for the cloud if exists
+  * introduction of a concept for parametrized operations - an operation which requires additional configuration or setting (eg c8y log request also requires supported log types)
+    * `add` param cloud_name parameter_name - adds single parameter to the list if doesn't exist
+    * `remove` param cloud_name parameter_name - remove single parameter to the list if doesn't exist
+    * `list` param [cloud_name]` - lists all parameter, unless specific cloud table name provided, then lists only parameters for the cloud if exists
```

Comment:
A proposal to manage operation parameter is to let the `tedge operations` command operate at 3 levels (cloud, operation, parameter), the latter being optional.

```
tedge operations add c8y c8y_SoftwareUpdate
tedge operations add c8y c8y_LogfileRequest software-management
tedge operations add c8y c8y_LogfileRequest mosquitto
tedge operations list
tedge operations list c8y
tedge operations list c8y c8y_LogfileRequest 
```